packages/app/src/update/services/device-update.service.ts

```diff
@@ -0,0 +1,218 @@
+  PhoneLockCategory,
+} from "App/device/constants"
+import { DeviceFileSystemService } from "App/device-file-system/services"
+import { DeviceInfo } from "App/device/types/mudita-os/serialport-request.type"
+
+// DEPRECATED
+import DeviceService from "App/__deprecated__/backend/device-service"
+
+export class DeviceUpdateService {
+  constructor(
+    private settingsService: SettingsService,
+    private deviceService: DeviceService,
+    private deviceFileSystem: DeviceFileSystemService
+  ) {}
+
+  public async updateOs(payload: UpdateOS): Promise<ResultObject<boolean>> {
+    const deviceInfoResult = await this.getDeviceInfo()
+
+    if (!deviceInfoResult.ok || !deviceInfoResult.data) {
+      return Result.failed(
+        new AppError(
+          UpdateError.CannotGetOsVersion,
+          "Current os version request failed"
+        )
+      )
+    }
+
+    const filePath = join(
+      this.settingsService.getByKey("osDownloadLocation") as string,
+      payload.fileName
+    )
+
+    const fileResponse = await this.deviceFileSystem.uploadFileLocally({
+      filePath,
+      targetPath: "/sys/user/update.tar",
+    })
+
+    if (!fileResponse.ok || !fileResponse.data) {
+      return Result.failed(
+        new AppError(
+          UpdateError.UpdateFileUpload,
+          `Cannot upload ${filePath} to device`
+        )
+      )
+    }
+
+    const pureUpdateResponse = await this.deviceService.request({
+      endpoint: Endpoint.Update,
+      method: Method.Post,
+      body: {
+        update: true,
+        reboot: true,
+      },
+    })
+
+    if (pureUpdateResponse.status !== RequestResponseStatus.Ok) {
+      return Result.failed(
+        new AppError(UpdateError.UpdateCommand, "Cannot restart device")
+      )
+    }
+
+    const deviceRestartResponse = await this.waitUntilDeviceRestart()
+
+    if (!deviceRestartResponse.ok) {
+      return deviceRestartResponse
+    }
+
+    if (
+      this.deviceService.currentDevice?.deviceType === DeviceType.MuditaPure
+    ) {
+      const deviceUnlockedResponse = await this.waitUntilDeviceUnlocked()
+
+      if (!deviceUnlockedResponse.ok) {
+        return deviceUnlockedResponse
+      }
+    }
+
+    const deviceInfoAfterUpdateResult = await this.getDeviceInfo()
+
+    if (!deviceInfoAfterUpdateResult.ok || !deviceInfoAfterUpdateResult.data) {
+      return Result.failed(
+        new AppError(
+          UpdateError.CannotGetOsVersion,
+          "New os version request failed"
+        )
+      )
+    }
+
+    const afterUpdateOsVersion = deviceInfoAfterUpdateResult.data.version
+    const beforeUpdateOsVersion = deviceInfoResult.data.version
+
+    if (beforeUpdateOsVersion === afterUpdateOsVersion) {
+      return Result.failed(
+        new AppError(
+          UpdateError.VersionDoesntChanged,
+          "The version OS isn't changed"
+        )
+      )
+    }
+
+    return Result.success(true)
+  }
+
+  private async getDeviceInfo(): Promise<ResultObject<DeviceInfo>> {
```

Comment:
Consider using `device-info.service.ts`

Reply:
let's leave it until the new device manager will be implemented. Because I'm predicting that it can cause complications during replacing the legacy `device-service` functionality with a new one.